viewer/src/main/webapp/viewer-html/components/OpenLayersSnappingController.js

```diff
@@ -0,0 +1,259 @@
+/*
+ * Copyright (C) 2015 B3Partners B.V.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+/**
+ *  @description Controller component for the Snapping control.
+ * @author <a href="mailto:markprins@b3partners.nl">Mark Prins</a>
+ *
+ * @class
+ */
+Ext.define("viewer.components.OpenLayersSnappingController", {
+    extend: "viewer.components.Component",
+    /**
+     * editable/drawable OpenLayers Vector layer.
+     * @private
+     * @TODO refactor to array as each edit & draw control have their own layer
+     */
+    frameworkLayer: null,
+    /** snapping control.*/
+    frameworkControl: null,
+    config: {
+        style: {
+            strokeColor: '#FF00FF',
+            strokeOpacity: 0.5,
+            strokeWidth: 1,
+            pointRadius: 1,
+            fillOpacity: 0.25,
+            fillColor: '#FF00FF'
+        },
+        viewerController: null
+    },
+    /**
+     * OpenLayers Vector layers to snap to.
+     * @private
+     */
+    snapLayers: [],
+    /**
+     * name prefix of the built-in snapLayers
+     */
+    snapLayers_prefix: "snapping_",
+    /**
+     * OpenLayers Map.
+     * @private
+     */
+    frameworkMap: null,
+    /**
+     * @constructor
+     * @param {Object} config
+     * @returns {viewer.viewercontroller.openlayers.OpenLayersSnappingLayer|OpenLayersSnappingControllerAnonym$0.constructor}
+     */
+    constructor: function (config) {
+        viewer.components.OpenLayersSnappingController.superclass.constructor.call(this, config);
+        this.frameworkMap = this.config.viewerController.mapComponent.getMap().getFrameworkMap();
```

Comment:
Overall note: 
It's too openlayers specific. I think it has to be made more framework independent, but you're already there.
The instantiation should be in OpenLayersMapComponent or OpenLayersVectorLayer (or somewhere else, but in a place which knows what framework is used).
I think it would also be good to make a superclass for the snappingcontroller, which has the basic implementation (like adding features to the vectorlayers, listening to the events, etc.), so the framework specific implementation should be quite limited and restricted to framework specific calls (and probably more of a passthrough for calls made by Snapping.js).
Framework specific files should be under it's framework folder for clarity.